stm32cube/stm32f7xx/CMakeLists.txt

```diff
@@ -25,7 +25,11 @@ zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DMA drivers/src/stm32f7xx_hal_
 zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DMA2D drivers/src/stm32f7xx_hal_dma2d.c)
 zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DMA_EX drivers/src/stm32f7xx_hal_dma_ex.c)
 zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DSI drivers/src/stm32f7xx_hal_dsi.c)
-zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH drivers/src/Legacy/stm32f7xx_hal_eth.c)
+if(CONFIG_ETH_STM32_HAL_API_V2)
+  zephyr_library_sources(drivers/src/stm32f7xx_hal_eth.c)
```

Comment:
could this be ?
`zephyr_library_sources_ifdef(CONFIG_ETH_STM32_HAL_API_V2 drivers/src/stm32f7xx_hal_eth.c)`
`zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH drivers/src/Legacy/stm32f7xx_hal_eth.c)`

Reply:
No, `CONFIG_USE_STM32_HAL_ETH` is enabled in both cases